crates/sui-transaction-builder/src/lib.rs

```diff
@@ -63,12 +63,12 @@ impl TransactionBuilder {
         &self,
         signer: SuiAddress,
         input_gas: Option<ObjectID>,
-        budget: u64,
+        gas_budget: u64,
         input_objects: Vec<ObjectID>,
         gas_price: u64,
     ) -> Result<ObjectRef, anyhow::Error> {
```

Comment:
As this is a private function, you can probably clean its interface up so that it 

- Takes a `Vec<ObjectID>` of input gas coins and returns a `Vec<ObjectRef>`.
- Accepts a `TransactionKind` instead of a list of input objects, and it does the input object calculation.

Reply:
The issue with this is that all the "original" tx related calls are using it, and those are used by the `json-rpc` so I am a bit scared in changing this. Most of the `TransactionData` functions only take a ObjectRef for gas payment, and not a vec, so not sure what's the improvement there.

Reply:
That would also require to create a `tx_kind` for all those original `tx` functions and pass that into the `select_gas`. Let's do this in a follow-up PR if we really want to.